client/src/app.d.ts

```diff
@@ -1,12 +1,19 @@
+import type { User as DomainUser } from "$shared/domain";
+
 // See https://kit.svelte.dev/docs/types#app
 // for information about these interfaces
 declare global {
 	namespace App {
+		interface Locals {
+			user: PublicUser
+		}
 		// interface Error {}
-		// interface Locals {}
 		// interface PageData {}
 		// interface Platform {}
 	}
 }
 
-export {};
+export type PublicUser = Omit<DomainUser, 'id'> | undefined | null
```

Comment:
Newline and also why is this important? Is this for development purposes or security as well. Because I would not think that Omit really omits anything in practice, only the type definition's field is omitted, not the actual runtime object field (I may be wrong though). So if it is for security then it can surely leak out, but maybe I am overthinking and this is not for security and now I made you read all this nonsense 😄